### PR TITLE
issue #27773 First pass at adding site restrictions

### PR DESCRIPTION
**Pull request by ChristopherCotnoir**

### CODE REVIEW

File: foundation-database/public/tables/metasql/address-uses.mql [hunk 1]
```diff
@@ -31,6 +31,6 @@ UNION
 SELECT warehous_id, 5, <? value("whs") ?>, warehous_code,
        warehous_descrip, '', '',
        '', '', '', ''
-FROM whsinfo
+FROM site()
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/address-uses.mql]: Change isn't necessary here. The user should be allowed to see the warehouse itself.

File: foundation-database/public/tables/metasql/contact-muststayactive.mql [hunk 2]
```diff
@@ -20,7 +20,7 @@ SELECT EXISTS(
   WHERE (vend_active
     AND (<? value("id") ?> IN (vend_cntct1_id, vend_cntct2_id)))
  UNION SELECT 7
-   FROM prospect
+   FROM prospect JOIN site() ON (prospect_warehous_id=warehous_id)
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/contact-muststayactive.mql]: This query checks whether it's safe to mark a Contact inactive. It should remain unchanged here and at line 33.

File: foundation-database/public/tables/metasql/crm_strictcountrycheck.mql [hunk 3]
```diff
@@ -61,7 +61,7 @@ SELECT SUM(counter) AS counter FROM (
   SELECT <? if exists("count") ?> COUNT(*)
          <? else ?> quhead_id, 'Quote', quhead_number, quhead_billtocountry, 'Bill-To'
          <? endif ?>
-    FROM quhead
+    FROM quhead JOIN site() ON (quhead_warehous_id=warehous_id)
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/crm_strictcountrycheck.mql]: This query is used to see if it's safe to impose a constraint on data entry. Please leave the old version.

File: foundation-database/public/tables/metasql/employees-detail.mql [hunk 4]
```diff
@@ -17,7 +17,7 @@ SELECT emp_id, warehous_code, emp_active, emp_code, emp_number,
 <? endforeach ?>
 FROM emp
   LEFT OUTER JOIN cntct ON (emp_cntct_id=cntct_id)
-  LEFT OUTER JOIN whsinfo ON (emp_warehous_id=warehous_id)
+  LEFT OUTER JOIN site() ON (emp_warehous_id=warehous_id)
```
Comment (gilmoskowitz) [inline foundation-database/public/tables/metasql/employees-detail.mql]: The net result of this change is that the same data will be returned but some rows won't have a `warehous_code`. Search for `join_type` in the [SELECT](https://www.postgresql.org/docs/9.3/static/sql-select.html) documentation.

**Comment by lokeshsh** [comment]
Performed UAT: Issue is fixed.

**Pull request closed without merging**